scripts/tests/avh/helpers/avh_instance.py

```diff
@@ -0,0 +1,240 @@
+
+        ssh_client = paramiko.SSHClient()
+        ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+        ssh_proxy_client.connect(
+            hostname=proxy_hostname,
+            username=proxy_username,
+            pkey=self.ssh_pkey,
+            look_for_keys=False,
+            timeout=timeout,
+        )
+
+        try:
+            proxy_sock = ssh_proxy_client.get_transport().open_channel(
+                kind="direct-tcpip",
+                dest_addr=(instance_ip, 22),
+                src_addr=("", 0),
+                timeout=timeout,
+            )
+
+            ssh_client.connect(
+                hostname=instance_ip,
+                username=self.username,
+                password=self.password,
+                sock=proxy_sock,
+                timeout=timeout,
+                look_for_keys=False,
+            )
+        except Exception:
+            raise Exception(
+                f"Failled to connect to {instance_ip} via SSH proxy {proxy_username}@{proxy_hostname}"
+            )
+
+        return ssh_client
+
+    def delete(self):
+        if self.console is not None:
+            self.console.close()
+
+            self.console = None
+
+        if self.ssh_key_id is not None:
+            self.avh_client.delete_ssh_project_key(self.ssh_key_id)
+
+            self.ssh_key_id = None
+
+        if self.instance_id is not None:
+            self.avh_client.delete_instance(self.instance_id)
+
+    def wait_for_state_deleted(self, timeout=60):
+        if self.instance_id is None:
+            return
+
+        start_time = time.monotonic()
+
+        while True:
+            try:
+                self.avh_client.instance_state(self.instance_id)
+            except avh_api.exceptions.NotFoundException:
+                break
+
+            if (time.monotonic() - start_time) > timeout:
+                raise Exception(
+                    f"Timedout waiting for instance id {self.instance_id} to be deleted"
+                )
+
+            time.sleep(1.0)
+
+        self.instance_id = None
+
+    def upload_application_binary(self, local_path, remote_path):
+        ssh_client = self.ssh_client()
+
+        stfp_client = ssh_client.open_sftp()
+        stfp_client.put(local_path, remote_path)
+        stfp_client.close()
+
+        ssh_client.exec_command(f"chmod +x {remote_path}")
+        ssh_client.close()
+
+    def wait_for_console_output(self, expected_output, timeout=10.0):
+        self.console.settimeout(1.0)
+
+        start_time = time.monotonic()
+
+        output = b""
+        while True:
+            if (time.monotonic() - start_time) > timeout:
+                raise Exception(
+                    f"Timed out waiting for {expected_output} in console output"
+                    f"Current output is {output}"
+                )
+
+            try:
+                output += self.console.recv()
+            except websocket.WebSocketTimeoutException:
+                pass
```

Comment:
should we log? Ignoring exceptions is generally a code-smell. At least a comment should explain why this is ok to ignore.

Reply:
Added comment